NwPluginAPI/Core/Player.cs

```diff
@@ -916,6 +1002,36 @@ public void SendBroadcast(string message, ushort duration, BroadcastFlags type =
 			Server.Instance.GetComponent<Broadcast>().TargetAddElement(ReferenceHub.characterClassManager.connectionToClient, message, duration, type);
 		}
 
+		/// <summary>
+		/// Sets the player server role. why group is private....
+		/// </summary>
+		/// <param name="name">Name of server role to set.</param>
+		/// <param name="group">Group to be set.</param>
+		public void SetServerRole(string name, UserGroup group)

```

Comment:
What's a possible use-case for this? This only updates the user groups cache, so the changes are temporary (will be removed on round restart, config reload) and won't even update the player role (if the player is already on the server it won't update their role).

Reply:
I know, I wasn't looking for a permanent one, if I wanted a permanent one I would make it write to a file, I was thinking more about using it in a DB that would check every time a player enters and give him the role and when the round is over he reconnects to the server so it would re-run the event.

Reply:
I use this method for the VIPs of my server, when a player enters I check if he is in the DB cache, I see what group he is in (vip, vip+ etc) and if he has not yet expired.

```cs
 public async void CheckForVIPAsync(Player player)
        {
            var role = TempRoleAPI.GetGroup(player.UserId);

            if (role != null)
            {
                if (ServerStatic.GetPermissionsHandler()._groups.ContainsKey(role))
                {
                    var group = ServerStatic.GetPermissionsHandler()._groups[role];

                    player.SetRank(group.BadgeText, group);
                }
                else
                {
                    Log.Error($"UserGroup {role} dont exist");
                }
            }
        }
```

Reply:
I think that an event that can override getting role/group will be better.

Reply:
Curiously I had already made a branch with new events and I wanted to add PlayerChangingGroup, but I have no idea if ServerRoles.SetGroup is executed every time someone connects or only when they have a UserGroup.
https://github.com/SrLicht/NwPluginAPI/commit/aafd758d00452b0a1aa5d2f84c85698cce4715ab#